TTS/tts/models/vits.py

```diff
@@ -1569,10 +1569,17 @@ def init_from_config(config: "VitsConfig", samples: Union[List[List], List[Dict]
         from TTS.utils.audio import AudioProcessor
 
         upsample_rate = torch.prod(torch.as_tensor(config.model_args.upsample_rates_decoder)).item()
+
         if not config.model_args.encoder_sample_rate:
             assert (
                 upsample_rate == config.audio.hop_length
             ), f" [!] Product of upsample rates must be equal to the hop length - {upsample_rate} vs {config.audio.hop_length}"
+        else:
+            encoder_to_vocoder_upsampling_factor = config.audio.sample_rate / config.model_args.encoder_sample_rate
+            effective_hop_length = config.audio.hop_length * encoder_to_vocoder_upsampling_factor
```

Comment:
For upsampling we have two differents mode:

1. Upsampling using vocoder
2. Upsampling using interpolation

I think that upsampling using interpolation will break here, right?  Upsampling using interpolation does not change the vocoder/decoder upsample_rates. 

Changes the 
`if not config.model_args.encoder_sample_rate:` to:
` if not config.model_args.encoder_sample_rate or (config.model_args.encoder_sample_rate and config.model_args.interpolate_z):` can solve this issue but we need to test it.



So

Reply:
"Upsampling using interpolation does not change the vocoder/decoder upsample_rates."

Why is that? In the end, decoder and vocoder outputs have different sampling rates right? If so, it needs to be indicated in the config.

Reply:
> "Upsampling using interpolation does not change the vocoder/decoder upsample_rates."
> 
> Why is that? In the end, decoder and vocoder outputs have different sampling rates right? If so, it needs to be indicated in the config.

I didn't get what you mean. The sampling rate of the encoder part and decoder (vocoder) are different if "model_args.encoder_sample_rate" is not None on the config. The encoder part will be trained using the "model_args.encoder_sample_rate" and the vocoder part with the audio config sampling rate. For the decoder part, to match the training sampling rate we can change the vocoder/decoder upsample_rates  (upsampling using vocoder) or just interpolate the 22khz "z" latent variable to match the desired decoder input (upsampling using interpolation). This "assert" statmentthat you did will break because the upsampling using interpolation does not change the decoder architecture.